engine/dlib/src/dlib/file_descriptor.cpp

```diff
@@ -0,0 +1,158 @@
+// Copyright 2021 The Defold Foundation
+// Licensed under the Defold License version 1.0 (the "License"); you may not use
+// this file except in compliance with the License.
+//
+// You may obtain a copy of the License, together with FAQs at
+// https://www.defold.com/license
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#include "file_descriptor.h"
+#include "log.h"
+#include <stdint.h>
+#include <assert.h>
+#if defined(_WIN32)
+#include <winsock2.h>
+#else
+#include <poll.h>
+#endif
+
+
+namespace dmFileDescriptor
```

Comment:
I'm not entirely sure about the naming here. This new namespace currently only contains functionality to use poll() on one or more file descriptors. Could we move this somewhere else?